lib/timeline/component/graph2d_types/points.js

```diff
@@ -31,8 +31,69 @@ Points.prototype.draw = function(dataset, group, framework, offset) {
  */
 Points.draw = function (dataset, group, framework, offset) {
   if (offset === undefined) {offset = 0;}
+  
+  var callback = undefined;
+  // check for the graph2d onRender
+  if (framework.options.drawPoints.onRender) {
+	  allback = framework.options.drawPoints.onRender;
+  }
+  
+  // override it with the group onRender if defined
+  if (group.options.drawPoints.onRender) {
+  	callback = group.options.drawPoints.onRender;
+  }
+
+  if (callback) {
+  	/*
+  	 * If a callback function is provided, it might return an object with drawing information about that point
+  	 * In that case we need to save the previously defined 'drawPoints' options from the group.options object
+  	 * in order to preseve the group options.
+  	 * 1. We want to avoid to clone the group object again and again within the for-loop, therefore not using deepExtend.
+  	 * 2. In addition, the method signature of DOMutil.drawPoint(..) should be left unchanged, otherwise we would add the callbackResult to the method signature.
+  	 *    This means we need to modify the group object passed into the DOMutil.drawPoint(..). and restore it afterwards.
+  	 */
+  	saveOptions(group.options);
+  }
+
   for (var i = 0; i < dataset.length; i++) {
-    DOMutil.drawPoint(dataset[i].x + offset, dataset[i].y, group, framework.svgElements, framework.svg, dataset[i].label);
+    if (!callback) {
+  	  // draw the point the simple way.
+  	  DOMutil.drawPoint(dataset[i].x + offset, dataset[i].y, group, framework.svgElements, framework.svg, dataset[i].label);
+    } else {
+  	  var callbackResult = callback(dataset[i], group, framework); // result might be true, false or an object
+  	  if (!callbackResult) { // false or null
+  		  continue;
+  	  }
+  	  
+   /*
+    * In case it is an object try to use the object provided parameters
+    * In case it is true, the group settings will be used
+    */
+   group.options.drawPoints = {
+  	  enabled: group.options.drawPoints.enabled,
+  	  onRender: group.options.drawPoints.onRender,
+  	  style: callbackResult.style || group.options.drawPoints.style,
+  	  size: callbackResult.size || group.options.drawPoints.size
+   };
+   DOMutil.drawPoint(dataset[i].x + offset, dataset[i].y, group, framework.svgElements, framework.svg, dataset[i].label);
```

Comment:
maybe we should change the arguments going into the drawPoint so we don't have to mess with the group options? We could make the arguments a JSON so it's more readable.